components/team_controller/index.ts

```diff
@@ -57,25 +58,25 @@ function mapStateToProps(state: GlobalState, ownProps: OwnProps) {
     };
 }
 
-function mapDispatchToProps(dispatch: Dispatch) {
-    return {
-        actions: bindActionCreators<ActionCreatorsMapObject<Action>, any>({
-            fetchMyChannelsAndMembers,
-            fetchAllMyTeamsChannelsAndChannelMembers,
-            getMyTeamUnreads,
-            viewChannel,
-            markChannelAsReadOnFocus,
-            getTeamByName,
-            addUserToTeam,
-            setPreviousTeamId,
-            selectTeam,
-            loadStatusesForChannelAndSidebar,
-            getAllGroupsAssociatedToChannelsInTeam,
-            getAllGroupsAssociatedToTeam,
-            getGroupsByUserIdPaginated,
-            getGroups,
-        }, dispatch),
-    };
-}
+const mapDispatchToProps = {
+    fetchMyChannelsAndMembers,
+    fetchAllMyTeamsChannelsAndChannelMembers,
+    getMyTeamUnreads,
+    viewChannel,
+    markChannelAsReadOnFocus,
+    getTeamByName,
+    addUserToTeam,
+    setPreviousTeamId,
+    selectTeam,
+    loadStatusesForChannelAndSidebar,
+    getAllGroupsAssociatedToChannelsInTeam,
+    getAllGroupsAssociatedToTeam,
+    getGroupsByUserIdPaginated,
+    getGroups,
+};
+
+const connector = connect(mapStateToProps, mapDispatchToProps);
+
+export type PropsFromRedux = ConnectedProps<typeof connector>;
 
-export default withRouter(connect(mapStateToProps, mapDispatchToProps)(NeedsTeam));
```

Comment:
Didn't you just add a `withRouter` on a different PR? I'm curious why we sometimes need it while other times we don't.

Reply:
Yes i added it for the `CenterChannel` because it is a class component, and for it to access match etc it needs withRouter hoc to access those props, But new team_controller is functional so i can access parms with hooks instead. that why removed the hoc